vulnerabilities/pipelines/pypa_importer.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import logging
+from pathlib import Path
+from typing import Iterable
+
+import saneyaml
+from fetchcode.vcs import fetch_via_vcs
+from fetchcode.vcs.git import Git
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importers.osv import parse_advisory_data
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipeline
+from vulnerabilities.utils import get_advisory_url
+
+module_logger = logging.getLogger(__name__)
+
+
+class PyPaImporterPipeline(VulnerableCodeBaseImporterPipeline):
+    """Collect advisories from PyPA GitHub repository."""
+
+    spdx_license_expression = "CC-BY-4.0"
+    license_url = "https://github.com/pypa/advisory-database/blob/main/LICENSE"
+    repo_url = "git+https://github.com/pypa/advisory-database"
+    importer_name = "PyPa Importer Pipeline"
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.clone,
+            cls.collect_and_store_advisories,
+            cls.import_new_advisories,
+            cls.clean_downloads,
+        )
+
+    def clone(self):
+        self.log(f"Cloning `{self.repo_url}`")
+        self.vcs_response = fetch_via_vcs(self.repo_url)
+
+    def advisories_count(self):
+        vulns_path = Path(self.vcs_response.dest_dir) / "vulns"
+        return sum(1 for _ in vulns_path.rglob("*.yaml"))
+
+    def collect_advisories(self) -> Iterable[AdvisoryData]:
+        base_directory = self.vcs_response.dest_dir
+        base_path = Path(base_directory)
+        vulns_path = base_path / "vulns"
+        revision = Git.get_revision(base_directory) or "main"
+
+        for advisory in vulns_path.rglob("*.yaml"):
+            advisory_url = get_advisory_url(
```

Comment:
get_advisory_url() could get a docstring... it's code is not obvious to me

Reply:
Done.